### DISCUSSION TITLE
GF dance fix

### DISCUSSION THREAD

**Discussion by shaderw0lf** [KadeArchive/Kade-Engine#627] [Show and tell]
I compiled Kade Engine and saw, what in tutorial here's strange bug with gf's dance (before starting song gf's dance is normal, but when playing song gf's dancing animation freezes in one direction).
So, I stupidly fixed it.
Maybe it will be useful for someone..?)

![изображение](https://user-images.githubusercontent.com/55358751/120088550-698ae780-c0fa-11eb-8a0a-3444adf6f6de.png)

Text version:

```
if (curSong == 'Tutorial' && dad.curCharacter == 'gf')
	{
		if (curBeat % 2 == 1 && dad.animOffsets.exists('danceLeft'))
			dad.playAnim('danceLeft');
		if (curBeat % 2 == 0 && dad.animOffsets.exists('danceRight'))
			dad.playAnim('danceRight');
	}

if (SONG.notes[Math.floor(curStep / 16)] != null)
	{
		if (SONG.notes[Math.floor(curStep / 16)].changeBPM)
		{
			Conductor.changeBPM(SONG.notes[Math.floor(curStep / 16)].bpm);
			FlxG.log.add('CHANGED BPM!');
		}
		// else
		// Conductor.changeBPM(SONG.bpm);

		// Dad doesnt interupt his own notes
		if (SONG.notes[Math.floor(curStep / 16)].mustHitSection && dad.curCharacter != 'gf')
			dad.dance();
	}
```

**Comment by Burger-Gaming** [comment]
someone might have removed that in a commit or something

**Comment by puyoxyz** [comment]
you can make a [pr](https://github.com/KadeDev/Kade-Engine/pulls) if you want to add it to the official source code!

**Reply by shaderw0lf** [reply]
thanks for advice!

**Comment by RealDealJinx** [comment]
Very off topic but how did you get the color to show on a .hx file within Sublime
